src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,37 @@
+package seedu.address.model.person;
+
+import java.util.Arrays;
+
+import java.util.List;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code ReadOnlyPerson}'s {@code Tags} matches any of the keywords given.
+ */
+
+public class TagContainsKeywordsPredicate implements Predicate<ReadOnlyPerson> {
+    private final List<String> keywords;
+
+    public TagContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(ReadOnlyPerson person) {
+        String tag = Arrays.toString(person.getTags().toArray())
+                .replaceAll("[\\[\\](),{}]", "");
+        return keywords.stream()
+                        .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(tag, keyword));
```

Comment:
the line wrap here doesn't follow our coding standard, indentation for wrapped lines should be only 8 spaces